ethcore/evm/src/instructions.rs

```diff
@@ -564,24 +571,26 @@ pub enum GasPriceTier {
 	Invalid
 }
 
-impl Default for GasPriceTier {
-	fn default() -> Self {
-		GasPriceTier::Invalid
+impl GasPriceTier {
+	/// Returns the index in schedule for specific `GasPriceTier`
+	pub fn idx(&self) -> usize {
```

Comment:
That "c-style enum" is actually just `GasPriceTier`. :)

We still need this `idx` function. It's used in `schedule`, where we dispatch based on the index to know about the default gas.

Reply:
Ah, didn't realize the values were assigned implicitly. In that case, seems like this method could just be implemented as `*self as usize` instead of the match, or inlined.

Reply:
I'm not really sure about that -- the performance is mostly same, but I'm a little bit concerned about clarity.

Unlike `Instruction`, We don't have enum-to-number natural corresponding for `GasPriceTier`. I tried adding the index to enum and make `GasPriceTier` a c-like enum. The issue, however, is that there're other number corresponding for `GasPriceTier` (like default gas cost). And the result might be a little bit confusing.